ghost/core/core/server/data/migrations/versions/5.21/2022-10-26-13-32-backfill-subscriptions.js

```diff
@@ -0,0 +1,112 @@
+const _ = require('lodash');
+const logging = require('@tryghost/logging');
+const ObjectId = require('bson-objectid').default;
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        // eslint-disable-next-line no-restricted-syntax
+        const members = await knex('members')
+            .select([
+                'members.id',
+                'members.status',
+                'members_products.product_id',
+                'members_products.expiry_at',
+                'members_stripe_customers_subscriptions.id AS mscs_id',
+                'members_stripe_customers_subscriptions.offer_id'
+            ])
+            .leftJoin('members_products', 'members.id', 'members_products.member_id')
+            .leftJoin('members_stripe_customers', 'members.id', 'members_stripe_customers.member_id')
+            .leftJoin('members_stripe_customers_subscriptions', 'members_stripe_customers.customer_id', 'members_stripe_customers_subscriptions.customer_id');
+
+        if (!members.length) {
+            logging.info(`No members found - skipping subscription backfill`);
+            return;
+        }
+
+        const freeProduct = await knex('products')
+            .select('id')
+            .where('type', 'free')
+            .first();
+
+        if (!freeProduct) {
+            logging.warn(`Unable to fetch free product - skipping subscription backfill`);
+            return;
+        }
+
+        const now = knex.raw('CURRENT_TIMESTAMP');
+
+        const ghostSubscriptionIds = [];
+
+        const rowsToInsert = members.map((m) => {
```

Comment:
I should probably consider looking at the `members_stripe_customers_subscriptions.type` first because we a free member might have a canceled paid subscription that we should also include